### PR TITLE
#4078 [runtime/java] fixed npe due to CommonToken.getText() returning null when inputStream is null

### PR DESCRIPTION
**Pull request by vityaman**
Solution of Issue 4078: https://github.com/antlr/antlr4/issues/4078

The problem was that CommonToken::getText() returns null if and only if it has InputStream field null, what happens when we call CommonToken(int type) as there source is equal to 
```
protected static final Pair<TokenSource, CharStream> EMPTY_SOURCE =
    new Pair<TokenSource, CharStream>(null, null);
```

```
// CommonToken.java

@Override
public String getText() {
	if ( text!=null ) {
		return text;
	}
	CharStream input = getInputStream();
	if ( input==null ) return null; // <-- oops
	int n = input.size();
	if ( start<n && stop<n) {
		return input.getText(Interval.of(start,stop));
	}
	else {
		return "<EOF>";
	}
}
// ...
public String toString(Recognizer<?, ?> r) {
	String channelStr = "";
	if ( channel>0 ) {
		channelStr=",channel="+channel;
	}
	String txt = getText();
	if ( txt!=null ) { // <-- this is what saved us in toString
		txt = txt.replace("\n","\\n");
		txt = txt.replace("\r","\\r");
		txt = txt.replace("\t","\\t");
	}
	else {
		txt = "<no text>";
	}
	String typeString = String.valueOf(type);
	if ( r!=null ) {
		typeString = r.getVocabulary().getDisplayName(type);
	}
	return "[@"+getTokenIndex()+","+start+":"+stop+"='"+txt+"'<"+typeString+">"+channelStr+","+line+":"+getCharPositionInLine()+"]";
}
```

### CODE REVIEW

File: runtime/CSharp/src/Atn/LexerATNSimulator.cs [hunk 1]
```diff
@@ -779,7 +779,7 @@ public int Column
 		}
 
 
-		public void Consume(ICharStream input)
+		public virtual void Consume(ICharStream input)
```
Comment (KvanTTT) [inline runtime/CSharp/src/Atn/LexerATNSimulator.cs:782]: It relates neither to PR nor to the Java target.

File: runtime/Java/src/org/antlr/v4/runtime/misc/Utils.java [hunk 2]
```diff
@@ -61,10 +61,10 @@ public  static <T> void removeAllElements(Collection<T> data, T value) {
...
 		StringBuilder buf = new StringBuilder();
 		for (char c : s.toCharArray()) {
-			if ( c==' ' && escapeSpaces ) buf.append('\u00B7');
-			else if ( c=='\t' ) buf.append("\\t");
-			else if ( c=='\n' ) buf.append("\\n");
-			else if ( c=='\r' ) buf.append("\\r");
+			if (c == ' ' && escapeSpaces) buf.append('\u00B7');
+			else if (c == '\t') buf.append("\\t");
+			else if (c == '\n') buf.append("\\n");
+			else if (c == '\r') buf.append("\\r");
```
Comment (KvanTTT) [inline runtime/Java/src/org/antlr/v4/runtime/misc/Utils.java:67]: Please don't mix whitespace and semantics changes (ideally, don't change whitespaces without need).

File: runtime/Java/src/org/antlr/v4/runtime/tree/TerminalNodeImpl.java [hunk 3]
```diff
@@ -60,8 +60,10 @@ public String toStringTree(Parser parser) {
 
 	@Override
 	public String toString() {
-			if ( symbol.getType() == Token.EOF ) return "<EOF>";
-			return symbol.getText();
+		if (symbol.getType() == Token.EOF) {
+			return "<EOF>";
+		}
+		return symbol.getText();
```
Comment (KvanTTT) [inline runtime/Java/src/org/antlr/v4/runtime/tree/TerminalNodeImpl.java:66]: Also, please revert this change.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

**Comment by vityaman** [comment]
Will close this and resolve issues in PR https://github.com/antlr/antlr4/pull/4252 as here I have problems with commit history and it is easier to solve them by creating a fresh one.
